pandas/core/internals/managers.py

```diff
@@ -350,25 +351,62 @@ def _verify_integrity(self) -> None:
                 f"tot_items: {tot_items}"
             )
 
-    def reduce(self, func, *args, **kwargs):
+    def reduce(self, func, ignore_failures=False):
         # If 2D, we assume that we're operating column-wise
         if self.ndim == 1:
             # we'll be returning a scalar
             blk = self.blocks[0]
-            return func(blk.values, *args, **kwargs)
+            return func(blk.values)
 
         res = {}
         for blk in self.blocks:
-            bres = func(blk.values, *args, **kwargs)
+            placement = blk.mgr_locs.as_array
+            if isinstance(blk, CategoricalBlock):
```

Comment:
(we could also have a `Block.reduce` that can hold those Block-specific logic), to keep this BlockManager.reduce more straightforward)